packages/dialect-react-ui/components/Chat/screens/Main/pages/CreateThreadPage/CreateThread.tsx

```diff
@@ -122,103 +123,162 @@ function ActionCaption({
-        Invalid address or Twitter handle
+      <P
+        className={clsx(textStyles.small, 'dt-text-green-500 dt-mt-1 dt-px-2')}
+      >
+        {address}
+      </P>
+    );
+  }
+
+  if (valid && twitterHandle && snsDomain) {
+    return (
+      <P
+        className={clsx(textStyles.small, 'dt-text-green-500 dt-mt-1 dt-px-2')}
+      >
+        SNS domain: {snsDomain}.sol / Twitter handle: {twitterHandle}
       </P>
     );
   }
 
-  if (!isTwitter && valid && !cardinalAddress) {
+  if (valid && snsDomain) {
     return (
       <P
         className={clsx(textStyles.small, 'dt-text-green-500 dt-mt-1 dt-px-2')}
       >
-        Valid address
+        SNS domain: {snsDomain}.sol
       </P>
     );
   }
 
-  if (!isTwitter && valid && cardinalAddress) {
+  if (valid && twitterHandle) {
     return (
       <P
         className={clsx(textStyles.small, 'dt-text-green-500 dt-mt-1 dt-px-2')}
       >
-        {cardinalAddress}
+        Twitter handle: {twitterHandle}
       </P>
     );
   }
+
+  return (
+    <P className={clsx(textStyles.small, 'dt-text-green-500 dt-mt-1 dt-px-2')}>
+      Valid address
+    </P>
+  );
+};
+
+const parseSNSDomain = (domainString: string) => {
+  domainString = domainString.trim();
+  const isSNSDomain = domainString.match(SNS_DOMAIN_EXT);
+  const domainName = domainString.slice(0, domainString.length - 4);
+  if (!isSNSDomain || !domainName) return;
+  return domainName;
 };
 
-const fetchSNSDomain = async (
+const tryFetchSNSDomain = async (
   connection: anchor.web3.Connection,
   domainName: string
 ) => {
-  const hashedName = await getHashedName(domainName);
+  try {
+    const hashedName = await getHashedName(domainName);
 
-  const domainKey = await getNameAccountKey(
-    hashedName,
-    undefined,
-    SOL_TLD_AUTHORITY
-  );
+    const domainKey = await getNameAccountKey(
+      hashedName,
+      undefined,
+      SOL_TLD_AUTHORITY
+    );
+
+    const { registry } = await NameRegistryState.retrieve(
+      connection,
+      domainKey
+    );
 
-  const { registry } = await NameRegistryState.retrieve(connection, domainKey);
+    return registry?.owner;
+  } catch (e) {}
+};
+
+const parseTwitterHandle = (handleString: string) => {
+  handleString = handleString.trim();
+  const isTwitter = handleString.startsWith('@');
+  const handle = handleString.substring(1, handleString.length);
+  if (!isTwitter || !handle) return;
+  return handle;
+};
+
+const tryFetchAddressFromTwitterHandle = async (
+  connection: anchor.web3.Connection,
+  handle: string
+) => {
+  try {
+    const { result } = await fetchAddressFromTwitterHandle(connection, handle);
+
+    return result?.parsed.data;
+  } catch (e) {}
+};
 
-  return registry?.owner;
+const tryPublicKey = (addressString: string) => {
+  let pubKey: anchor.web3.PublicKey | null = null;
+  try {
+    pubKey = new anchor.web3.PublicKey(addressString);
+    return pubKey;
```

Comment:
shouldn't we always return pubKey? so the function return null in case if error is thrown

Reply:
yeah, good point, updated it